mesonbuild/compilers/c.py

```diff
@@ -526,9 +528,9 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
                            full_version=full_version)
         ClangClCompiler.__init__(self, target)
 
-    def get_option_compile_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
-        key = self.form_compileropt_key('std')
-        std = options.get_value(key)
+    def get_option_compile_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
```

Comment:
The file already uses string types in many places. Harmonizing should be done separately.

Reply:
There is no useful argument for harmonizing on string types just because the file already uses it in many places. When we write new code, we should be naturally using the newer and more compact form, so that over time as code is rewritten in the natural course of development, the more wordy, less space efficient, older style disappears without having to have dedicated commits that just change tons of strings we recently introduced.

You wouldn't say "the file already doesn't use type annotations in many places. Harmonizing on the use of type annotations should be done separately". In fact, @dcbaker and I want exactly the opposite of harmonizing, if you think about it. :) We would like new code to look different from old code, in the same file...